alarm-frontend/src/Components/Alarm.js

```diff
@@ -1,63 +1,67 @@
-import React, { useContext } from "react";
-import { minutesNumber, hourNumber } from "../fixNumbers";
-import useOption from "../Hooks/useOption";
+import React, { useContext, useEffect, useState } from "react";
 import { Context } from "./Context";
+import '../css/alarm.css';
+import Sound from "../LostWoods.mp3";
+import axios from "axios";
+const BASE_URL = "http://127.0.0.1:5000"
 
-function Alarm() {
-    const [hour, setHour] = useOption("Hour");
-    const [minutes, setMinutes] = useOption("Minutes");
-    const [amPmOption, setAmPmOption] = useOption("Am-Pm");
-    const { setAlarmTime, pauseAlarm, alarmExists, setAlarmExists } =
-        useContext(Context);
-
+function Alarm({AMorPM, difficulty, hour, id, minute, type, handler}) {
+    const [alarm] = useState(new Audio(Sound))
+    const { realHour, realMinute, amPm } = useContext(Context);
+    const [alarmTime, setAlarmTime] = useState();
+    const [alarmExists, setAlarmExists] = useState("");
 
     const setAlarm = () => {
+        console.log("You have set an alarm!")
         setAlarmExists(true);
-        setAlarmTime(`${hour}${minutes}${amPmOption}`)
+        setAlarmTime(`${hour}${minute}${AMorPM}`);
     };
 
-    const stopAlarm = () => {
-            pauseAlarm();
-            setAlarmExists(false);
+    const noAlarm = () => {
+        console.log("Alarm is turned off!")
+        setAlarmExists(false);
+        setAlarmTime("");
+    }
+
+    const pauseAlarm = () => {
+        console.log("Alarm is paused")
+        alarm.pause();
+    }
+
+    async function deleteAlarm() {
+        await axios.post(`${BASE_URL}/alarms/${id}/delete`);
+        handler();
+    }
+
+    if (hour < 10) hour = `0${hour}`
+    if (minute < 10) minute = `0${minute}`
```

Comment:
Make sure magic numbers are stored in clearly named constants. Also maybe have this logic in a function so you can test it later.